agent/config/config.go

```diff
@@ -327,8 +359,9 @@ func Default() (*Agent, error) {
 	}
 
 	return &Agent{
-		LogLevel:  defaultLogLevel,
-		PluginDir: pwd + defaultPluginDirSuffix,
+		LogLevel:                 defaultLogLevel,
+		PluginDir:                pwd + defaultPluginDirSuffix,
+		DynamicApplicationSizing: &DynamicApplicationSizing{},
```

Comment:
no defaults for these?

Reply:
I thought it would be better to set the defaults in the ENT repo, so for OSS this config block is completely ignored. 

The parsing of config and CLI flags is done here to avoid errors if an OSS binary is used when loading a config that has ENT values. 

We could check if ENT-only configs are used and emit a warning or exit early with a nicer error message.